scripts/upgrade.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+SUFFIX=`date +%s | shasum | base64 | head -c 8 ; echo`
+KEY=
+UPGRADE_FROM=juno
+UPGRADE_TO=master
+
+usage()
+{
+    echo Usage: $0 "--key <key-name>" "--from <release-name>" "--to <release-name>" "--suffix <instance-name-suffix>" >&2
+    echo "Options:" >&2
+    echo "    -k | --key    The name of the public key to use in all the instances." >&2
+    echo "    -f | --from   The OpenStack release to start from. (defaults to juno)" >&2
+    echo "    -t | --to     The OpenStack release to upgrade to (defaults to master)." >&2
+    echo "    -s | --suffix A string to append to all instance names (defaults to a randomly generated string)." >&2
+    exit 1
+}
+
+while [[ $# > 0 ]]; do
+    opt="$1"
+    case $opt in
+        -k|--key)
+            KEY="$2"
+            shift
+            ;;
+        -f|--from)
+            UPGRADE_FROM="$2"
+            shift
+            ;;
+        -t|--to)
+            UPGRADE_TO="$2"
+            shift
+            ;;
+        -s|--suffix)
+            SUFFIX="$2"
+            shift
+            ;;
+        -h|--help)
+            usage
+            ;;
+        *)
+            break
+            ;;
+    esac
+    shift
+done
+
+if [ "$KEY" == "" ]; then
+    echo Error: Key must be specified.
+fi
+if [ "$UPGRADE_FROM" == "" ]; then
+    echo Error: Release to upgrade from must be specified.
+fi
+if [ "$UPGRADE_TO" == "" ]; then
+    echo Error: Release to upgrade to must be specified.
+fi
+if [ "$SUFFIX" == "" ]; then
+    echo Error: Suffix must be specified.
+fi
+if [ "$KEY" == "" -o "$UPGRADE_FROM" == "" -o "$UPGRADE_TO" == "" -o "$SUFFIX" == "" ]; then
+    usage
+fi
+
+# ensure heat and nova are present
+which heat >/dev/null 2>&1
+HEAT_PRESENT=$?
+which nova >/dev/null 2>&1
+NOVA_PRESENT=$?
+if [[ "$HEAT_PRESENT" != "0" && "$NOVA_PRESENT" != "0" ]]; then
+    echo Error: The heat and nova clients are not available. Run \"pip install python-heatclient python-novaclient\" to install them.
+    exit 1
+elif [[ "$HEAT_PRESENT" != "0" ]]; then
+    echo Error: The heat client is not available. Run \"pip install python-heatclient\" to install it.
+    exit 1
+elif [[ "$NOVA_PRESENT" != "0" ]]; then
+    echo Error: The nova client is not available. Run \"pip install python-novaclient\" to install it.
+    exit 1
+fi
+
+START=$(date +%s);
+
+printf "Deploying RPC $UPGRADE_FROM release with stack name rpc-$SUFFIX.\n"
+heat stack-create -f rpc_multi_node.yml rpc-$SUFFIX -P "heat_git_version=$UPGRADE_FROM;os_ansible_git_version=$UPGRADE_FROM;deploy_monitoring=no;key_name=$KEY;cluster_prefix=rpc-$SUFFIX" -t 150 >/dev/null
+
```

Comment:
I would argue that the heat_git_version should always be juno in this script, and not tied to the $UPGRADE_FROM variable, because if I want to upgrade from 10.1.15 (os_ansible_git_version), then this breaks since there is no 10.1.15 version of rpc-heat.